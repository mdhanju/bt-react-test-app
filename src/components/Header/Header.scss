header {
    &#btrc-Header {
        background-color: $bt-blue;
        color: $white;
        height: 8.7rem;
        min-height: 8.7rem;
        display: flex;
        align-items: center;
        align-content: center;

        .container-fluid {
            display: flex;
            justify-content: space-between;
            align-items: flex-end;
            flex-wrap: nowrap;

            .logo {
                .mobile-logo {
                    width: 3.2rem;
                    height: 3.2rem;
                }
            }

            @include media-breakpoint-down(sm) {
                padding: 0;
            }
        }

        .logo {
            display: flex;
            align-items: baseline;

            @include media-breakpoint-down(sm) {
                align-items: center;
                margin-left: 1.6rem;
            }

            .app-name {
                margin-left: 2rem;
                font-size: 2rem;
                letter-spacing: -0.032rem;

                @include media-breakpoint-down(sm) {
                    font-size: 1.8rem;
                }
            }
        }
    }
}
