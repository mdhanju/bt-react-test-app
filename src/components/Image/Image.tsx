import React from 'react'
import classnames from 'classnames'

const Image = ({
    className,
    src,
    alt,
    ...props
}: Props): React.ReactElement => (
    <img src={src} className={classnames(className)} alt={alt} {...props} />
)

type Props = {
    className?: string;
    src: string;
    alt?: string;
};

export default Image
