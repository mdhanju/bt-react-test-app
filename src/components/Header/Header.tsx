import React from 'react'
import classnames from 'classnames'
import desktopLogoSrc from '../../assets/images/logo-white.png'
import mobileLogoSrc from '../../assets/images/bt-icon.svg'
import { FormattedMessage } from 'react-intl'
import Image from '../Image'

const Header = ({ className, id, ...props }: Props): React.ReactElement => (
    <header id={id} className={classnames(className)} {...props}>
        <div className="container-fluid">
            <div className="logo">
                <Image
                    src={desktopLogoSrc}
                    alt="brand logo"
                    className="mobile-hide"
                />
                <Image
                    src={mobileLogoSrc}
                    alt="mobile brand logo"
                    className="mobile-logo desktop-hide"
                />
                <span className="app-name">
                    <FormattedMessage id="app.name" />
                </span>
            </div>
        </div>
    </header>
)

type Props = {
    id?: string;
    className?: string;
};

Header.defaultProps = {
    id: 'btrc-Header'
}

export default Header
