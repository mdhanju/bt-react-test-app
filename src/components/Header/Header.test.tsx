/* eslint-disable react/jsx-no-undef */

import React from 'react'
import Header from './Header'
import Image from '../Image'

describe('<Header />', () => {
    it('should render properly without crashing', () => {
        const wrapper = shallow(<Header />)
        expect(wrapper.exists()).toBe(true)
        expect(wrapper.getElements()).toMatchSnapshot()
        expect(wrapper.find(Image)).toHaveLength(2)
    })

    it('render the localized (En) Header', () => {
        const wrapper = render(
            <Intl locale="en">
                <Header />
            </Intl>
        )
        expect(wrapper.text()).toBe(`Sample App`)
        expect(wrapper.html()).toMatch(
            '<img src="logo-white.png" class="mobile-hide" alt="brand logo">'
        )
        expect(wrapper.html()).toMatch(
            '<img src="bt-icon.svg" class="mobile-logo desktop-hide" alt="mobile brand logo">'
        )
    })
})
