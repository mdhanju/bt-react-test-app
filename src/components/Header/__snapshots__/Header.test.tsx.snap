// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render properly without crashing 1`] = `
Array [
  <header
    className=""
    id="btrc-Header"
  >
    <div
      className="container-fluid"
    >
      <div
        className="logo"
      >
        <Image
          alt="brand logo"
          className="mobile-hide"
          src="logo-white.png"
        />
        <Image
          alt="mobile brand logo"
          className="mobile-logo desktop-hide"
          src="bt-icon.svg"
        />
        <span
          className="app-name"
        >
          <FormattedMessage
            id="app.name"
            values={Object {}}
          />
        </span>
      </div>
    </div>
  </header>,
]
`;
